Metadata/CIP-Certification-Metadata.md

```diff
@@ -0,0 +1,180 @@
+
+The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification provider, DApp developer, DApp and DApp version.
+
+### Use-cases and Stakeholders
+DApp developers will seek certification from one of the providers, according to their desired level of certification and will refer to the certificate on several platforms (e.g. on their website or in their DApp registration on a DApp store).
+
+Certification is to be provided by certification providers including: testing services (level 1), auditors (level 2), and verification services (level 3). Certification providers will issue these certificates refering to a particular version of a DApp that has succesfully gone through a certification process. 
+
+DApp stores and light wallets will be able to pull DApps information from DApps registration or chain exploring and will link a DApp to a corresponding set of certificates.
+
+End-user will interact with a DApp through a wallet and will be able to check the different certificates obtained by the DApp.
+
+
+## Specification
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. -->
+
+### Certification providers
+Certification providers will present which level of certification was reached and present evidence of the work done. Certification providers will sign the certificate to attest that they have done the work and prevent certificate forgeries.
+
+
+### Definitions
+- **dApp** - A decentralised application that is described by the combination of metadata, certificate and a set of used scripts.
+- **dApp Store** - A dApp aggregator application which follows on-chain data looking for and verifying dApp metadata claims, serving their users linked dApp metadata.
+
+
+### Properties
+
+**subject**, an UTF-8 encoded string, is a mandatory field which is an identifier of the claim subject.
+
+**certLevel**, an integer between 1 and 3, is a mandatory field that represents the level of certification reached in a certificate.
+
+**certificateIssuer**, a string, is a mandatory field that represents the company that has issued the certificate. It will be used with a public list of public keys to ensure that the certificate originates from a trusted certification provider. 
+
+**reportURLs**, an array of URLs, is a mandatory field that link to the actual certification report for anyone to read. This ensures transparency in the findings, what was and was not considered in the certification process.
```

Comment:
```suggestion
**reportURLs**, an array of URLs, is a field where each link points to the same actual certification report for anyone to read. This ensures transparency in the findings, what was and was not considered in the certification process.
```
Maybe this should not be mandatory. There might not be a report (at all, or yet), depending on the contract between the dApp developer and the certification party. Additionally, links can die.

Reply:
I don't think there could be a certification with no kind of report to be honnest but I can move that to non-mandatory for the moment.